classes/py1.html

```diff
@@ -62,6 +65,43 @@ <h2><a class='py1-link' href="{{ lesson.url }}">{{ lesson.title }}</a></h2>
                     <p class="py1-lesson-duration">{{ lesson.length }} min</p>
                 </div>
             </div>
+            {% endif %}
+            {% endfor %}
+        </div>
+
+        <!-- flow lesson cards -->
+        <h2 class = "title">flow</h2>
+        <div class='py1-lessons-wrapper'>
+            {% for lesson in site.py1 %}
+            {% if lesson.group == "flow" %}
+            <div class='py1-lesson'>
+                <div class='py1-lesson-title-bar'>
+                    <h2><a class='py1-lesson-link' href="{{ lesson.url }}">{{ lesson.title }}</a></h2>
+                    <p class='py1-step-number'>#{{ lesson.num }}</p>
+                </div>
+                <div class='py1-lesson-body'>
+                    <ul class='py1-list py1-lesson-goal-list'>
+                        {% for brief in lesson.agenda %}
+                        <li class='py1-lesson-goal'><span>{{ brief.text }}</span></li>
+                        {% endfor %}
+                    </ul>
+                </div>
+                <div class='py1-lesson-action-bar'>
+                    <div class='py1-action-group'>
+                        {% if lesson.slides_link %}
+                        <a class='py1-lesson-action' href="{{ lesson.slides_link }}">slides</a>
+                        {% endif %}
+                        {% if lesson.video_link %}
+                        <a class='py1-lesson-action' href="{{ lesson.video_link }}">video</a>
+                        {% endif %}
+                        {% if lesson.worksheet_link %}
+                        <a class='py1-lesson-action' href="{{ lesson.worksheet_link }}">worksheet</a>
+                        {% endif %}
+                    </div>
+                    <p class="py1-lesson-duration">{{ lesson.length }} min</p>
+                </div>
+            </div>
+            {% endif %}
```

Comment:
This is quite a bit of duplicated code with the previous section! Unfortunately, we can't really reformat the way the data is stored (as it is a collection), but we could instead create some sort of reusable component to represent this card. 

The Jekyll-based way we could do this is with an `_includes` "component", which we can then pass in information to render the card. I believe Jun has created something similar in [`py1-lesson-card-content`](https://github.com/uclaacm/teach-la-website/blob/master/_includes/py1-lesson-card-content.html), which you can check out! If you want an example of an include designed around parameters, you can check out [`blog-author-card.html`](https://github.com/uclaacm/teach-la-website/blob/master/_includes/blog-author-card.html); [here are the relevant docs on includes](https://jekyllrb.com/docs/includes/).

Reply:
Thanks so much! I was having trouble figuring out how to consolidate the duplicated code and just tried creating an `_includes` that consists of all the duplicated code and a parameter specifying the lesson group.